src/Uno/Prism.Uno/PrismApplicationBase.cs

```diff
@@ -0,0 +1,203 @@
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Prism
+{
+    /// <summary>
+    /// Base application class that provides a basic initialization sequence
+    /// </summary>
+    /// <remarks>
+    /// This class must be overridden to provide application specific configuration.
+    /// </remarks>
+    public abstract class PrismApplicationBase : Application
+    {
+        IContainerExtension _containerExtension;
+        IModuleCatalog _moduleCatalog;
+
+        /// <summary>
+        /// The dependency injection container used to resolve objects
+        /// </summary>
+        public IContainerProvider Container => _containerExtension;
+
+        protected override void OnLaunched(LaunchActivatedEventArgs args)
+        {
+            base.OnLaunched(args);
+            InitializeInternal();
+        }
+
+        /// <summary>
+        /// Run the initialization process.
+        /// </summary>
+        void InitializeInternal()
+        {
+            ConfigureViewModelLocator();
+            Initialize();
+            OnInitialized();
+        }
+
+        /// <summary>
+        /// Configures the <see cref="Prism.Mvvm.ViewModelLocator"/> used by Prism.
+        /// </summary>
+        protected virtual void ConfigureViewModelLocator()
+        {
+            ViewModelLocationProvider.SetDefaultViewModelFactory((view, type) =>
+            {
+                return Container.Resolve(type);
+            });
+        }
+
+        /// <summary>
+        /// Runs the initialization sequence to configure the Prism application.
+        /// </summary>
+        public virtual void Initialize()
+        {
+            ContainerLocator.SetContainerExtension(CreateContainerExtension);
+            _containerExtension = ContainerLocator.Current;
+            _moduleCatalog = CreateModuleCatalog();
+            RegisterRequiredTypes(_containerExtension);
+            RegisterTypes(_containerExtension);
+            _containerExtension.FinalizeExtension();
+
+            ConfigureModuleCatalog(_moduleCatalog);
+
+            var regionAdapterMappins = _containerExtension.Resolve<RegionAdapterMappings>();
+            ConfigureRegionAdapterMappings(regionAdapterMappins);
+
+            var defaultRegionBehaviors = _containerExtension.Resolve<IRegionBehaviorFactory>();
+            ConfigureDefaultRegionBehaviors(defaultRegionBehaviors);
+
+            RegisterFrameworkExceptionTypes();
+
+            var shell = CreateShell();
+            if (shell != null)
+            {
+                _containerExtension.Resolve<IRegionNavigationService>().NavigationFailed += (s, e) => Console.WriteLine($"Region navigation failed {e.Error}");
+                RegionManager.SetRegionManager(shell, _containerExtension.Resolve<IRegionManager>());
+                RegionManager.UpdateRegions();
+                InitializeShell(shell);
+            }
+
+            InitializeModules();
+
+            Suspending += (s, e) => OnSuspending(e);
```

Comment:
With this being called from the OnLaunched... can OnLaunched be called more than once during the App's Lifetime? i.e. the app is backgrounded momentarily and then restored? If so then this would be wiring this up multiple time... seems this might be better in the PrismApplicationBase ctor?

Reply:
Definitely is, my mistake!